sys/net/application_layer/gcoap/forward_proxy.c

```diff
@@ -45,6 +47,9 @@ typedef struct {
     event_t event;
 } client_ep_t;
 
+static char _forward_proxy_thread[PROXY_STACK_SIZE];
+static kernel_pid_t _forward_proxy_pid;
```

Comment:
IMHO, this should be in an optional sub-module (e.g. `forward_proxy_thread.c`). For an unencrypted proxy the extra thread is not needed, so this memory usage should not be pulled in.